src/main/java/pl/wojciechkostecki/revel/service/MenuItemService.java

```diff
@@ -0,0 +1,55 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.mapper.MenuItemMapper;
+import pl.wojciechkostecki.revel.model.Menu;
+import pl.wojciechkostecki.revel.model.MenuItem;
+import pl.wojciechkostecki.revel.model.dto.MenuItemDTO;
+import pl.wojciechkostecki.revel.repository.MenuItemRepository;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+@Service
+@Transactional
+public class MenuItemService {
+    private final MenuItemRepository menuItemRepository;
+    private final MenuService menuService;
+    private final MenuItemMapper itemMapper;
+
+    public MenuItemService(MenuItemRepository menuItemRepository, MenuService menuService, MenuItemMapper itemMapper) {
+        this.menuItemRepository = menuItemRepository;
+        this.menuService = menuService;
+        this.itemMapper = itemMapper;
+    }
+
+    public MenuItem save(MenuItemDTO menuItemDTO){
+        MenuItem menuItem = itemMapper.toEntity(menuItemDTO);
+        Menu menu = menuService.findById(menuItemDTO.getMenuId());
```

Comment:
I think you can do this in the mapper. `menuId` map to `menu.id` and that should be enough to connect them. I might be wrong though, but I think I've seen this in my work app.

Reply:
I can't find the solution you suggested. I tried to implement one suggestion from stackoverflow, but it didn't work and it took up more lines of code.